e2e/src/account/app.e2e-spec.ts

```diff
@@ -0,0 +1,62 @@
+/**
+ * @license
+ * Copyright (c) 2018 THREEANGLE SOFTWARE SOLUTIONS SRL
+ * Available under MIT license webFrame/LICENSE
+ */
+
+/**
+ * End-To-End tests for AppPage.
+ */
+import { IAccountCredentials } from 'app-shared/core';
+import { browser, logging } from 'protractor';
+
+import { AppPage } from './app.page';
+
+describe('Test if application pages work', (): void => {
+  let page: AppPage;
+
+  beforeEach((): void => {
+    page = new AppPage();
+  });
+
+  it('The "Login" public page should work', async (): Promise<void> => {
+    await page.openLoginPage();
+    expect(await page.getPageTitle()).toEqual('Login');
+  });
+
+  it('The "Signup" public page should work', async (): Promise<void> => {
+    await page.openSignupPage();
+    expect(await page.getPageTitle()).toEqual('Signup');
+  });
+
+  it('The "Forgot password" public page should work', async (): Promise<void> => {
+    await page.openForgotPasswordPage();
```

Comment:
Behind the scenes, you are testing if you navigate to a certain URL that it opens the expected page.
This is one scenario.
We should also test when you click on the forgot password button that you navigated to the correct page. This is not covered by the current tests.

```
LoginPageTestability
  clickForgotPassword()
  clickSignUp()
```

Reply:
One of the test suites that we have is now checking if the public `account/` pages can be opened via direct URL access and it's also checking if all the pages can be discovered by clicking on stuff.

P.S. With this new feature I had to remove the `autofocus` directive from the elements, since the tests were failing because some strange issues with change detection being triggered at the same time that we try to click on the "Forgot password?" link.